docs/design/AnoncredsW3CCompatibility.md

```diff
@@ -0,0 +1,286 @@
+
+### How a W3C credential is stored in the wallet.
+
+Storing a credential in the wallet is somewhat dependent on the kinds of metadata that are relevent. The metadata mapping between the W3C credential and an anoncreds credential is not fully clear yet.
+
+One of the questions we need to answer is whether the preferred approach is to modify the existing store credential function so that any credential type is a valid input, or whether there should be a special function just for storing W3C credentials.
+
+We will duplicate this [store_credential](https://github.com/hyperledger/aries-cloudagent-python/blob/8cfe8283ddb2a85e090ea1b8a916df2d78298ec0/aries_cloudagent/anoncreds/holder.py#L167) function and modify it:
+
+```
+async def store_w3c_credential(...) {
+    ...
+    ...
+    try:
+        cred = W3CCredential.load(credential_data)
+    ...
+    ...
+}
+```
+
+**Question: Would it also be possible to generate the credentials on the fly to eliminate the need for storage?**
+
+### How can we handle multiple signatures on a W3C VC Format credential?
+
+Only one of the signature types (CL) is allowed in the AnonCreds format, so if a W3C VC is created by `to_legacy()`, all signature types that can't be turned into a CL signature will be dropped. This would make the conversion lossy. Similarly, an AnonCreds credential carries only the CL signature, limiting output from `to_w3c()` signature types that can be derived from the source CL signature. A possible future enhancement would be to add an extra field to the AnonCreds data structure, in which additional signatures could be stored, even if they are not used. This could eliminate the lossiness, but it adds extra complexity and may not be worth doing.
+
+### Compatibility with AFJ: how can we make sure that we are compatible?
+
+We will write a test for the Aries Agent Test Framework that issues a W3C VC instead of an AnonCreds credential, and then run that test where one of the agents is ACA-PY and the other is based on AFJ -- and vice versa. Also write a test where a W3C VC is presented after an AnonCreds issuance, and run it with the two roles played by the two different agents. This is a simple approach, but if the tests pass, this should eliminate almost all risk of incompatibility.
+
+### Will we introduce new dependencies, and what is risky or easy?
+
+Any signfiicant bugs in the Rust implementation may prevent our wrappers from working, which would also prevent progress (or at least confirmed test results) on the higher-level code.
+
+If AFJ lags behind in delivering equivalent functionality, we may not be able to demonstrate compatibility with the test harness.
+
+### Where should the new issuance code go?
+
+So the [vc](https://github.com/hyperledger/aries-cloudagent-python/tree/main/aries_cloudagent/vc) directory contains code to verify vc's, is this a logical place to add the code for issuance?
+
+### What do we call the new things? Flexcreds? or just W3C_xxx
```

Comment:
I think AnonCreds in W3C VC (and/or VP) Format.  Long, ugly, but accurate.  AWVCF?  

FlexCreds is not really the right term, since it refers specifically to a VC with multiple attached signatures.

Reply:
If I put on my political hat, the naming strategy I'd be tempted to follow is to put the VC part first: W3C VC+Anon. You could also do something a bit like movies that use phrases like "The Matrix Reloaded" and do "VC Anonymized".

There are two reasons to put the VC part first. One is to make it impossible to refer to the technology without the VC brand (something that critics of Anoncreds have long urged people to do). The other is to make it clear that Anoncreds bring a new value to the VC ecosystem, by creating a minimal pair that highlights the anonymity. The thinking you want to encourage in a casual encounter is: "Do you want just plain VC, or do you want VC+Anon, which clearly has more features?"

Reply:
this makes sense to me - @swcurran if you agree we could try to make it compatible between the workstreams, @TimoGlastra if you agree?